source/vstd/std_specs/vec.rs

```diff
@@ -207,16 +215,37 @@ pub fn ex_vec_split_off<T, A: Allocator + core::clone::Clone>(
     vec.split_off(at)
 }
 
+pub open spec fn vec_clone_trigger<T, A: Allocator>(v1: Vec<T, A>, v2: Vec<T, A>) -> bool {
+    true
+}
+
 #[verifier::external_fn_specification]
 pub fn ex_vec_clone<T: Clone, A: Allocator + Clone>(vec: &Vec<T, A>) -> (res: Vec<T, A>)
     ensures
         res.len() == vec.len(),
-        forall |i| #![all_triggers] 0 <= i < vec.len() ==>
-            call_ensures(T::clone, (&vec[i],), res[i])
+        forall|i|
+            #![all_triggers]
+            0 <= i < vec.len() ==> call_ensures(T::clone, (&vec[i],), res[i]),
+        vec_clone_trigger(*vec, res),
+        vec@ =~= res@ ==> vec@ == res@,
 {
     vec.clone()
 }
 
+/*
+//TODO: improve pruning so that this is pruned away unless vec_clone_trigger is used
```

Comment:
Is this commented out because it would provide equality of deep view if we can prove extensional equality of the deep view of the vectors in the presence of the right trigger on `clone`, but you don't want it broadcasted everywhere?

Reply:
It's commented out because it would bloat the SMT encoding by bringing in everything about all the types implementing DeepView, even if the code never uses DeepView.  I would like to improve pruning of broadcasts soon so that it's based it on the triggers in each broadcast rather than always bringing in all broadcasts in the module.